src/amazon/request.ts

```diff
@@ -81,21 +80,29 @@ export class Request {
                 }
                 else {
                     // Expect content to be xml (content-type is not specified in every case)
-                    xmlParse(body, { explicitArray: false }, function(err, result) {
-                        if (err) {
-                            // Catch error at XML parsing
-                            callback({ origin: 'XMLParsing', message: err });
-                        } else if (_.has(result, 'ErrorResponse')) {
-                            // Catch error returned from API
-                            callback({ origin: 'MWS', message: result['ErrorResponse']['Error']['Message'], metadata: result['ErrorResponse']['Error'] });
-                        } else {
-                            // Return parsed result
-                            callback(null, result);
-                        }
-                    });
+                    // Catch uncaught errors from xml parsing lib
+                    try {
+                        var parser = new xml2js.Parser({ explicitArray: false });
```

Comment:
better move this line to constructor and make it a private variable or even better use dependency injection